src/lib/swapper/swappers/OsmosisSwapper/endpoints.ts

```diff
@@ -0,0 +1,210 @@
+import { cosmosChainId, osmosisChainId } from '@shapeshiftoss/caip'
+import type {
+  cosmos,
+  CosmosSdkChainAdapter,
+  GetFeeDataInput,
+  osmosis,
+} from '@shapeshiftoss/chain-adapters'
+import { bnOrZero } from '@shapeshiftoss/chain-adapters'
+import type { CosmosSignTx } from '@shapeshiftoss/hdwallet-core'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { Result } from '@sniptt/monads'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { v4 as uuid } from 'uuid'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type {
+  GetTradeQuoteInput,
+  GetUnsignedTxArgs,
+  SwapErrorRight,
+  Swapper2Api,
+  TradeQuote2,
+} from 'lib/swapper/api'
+import { SwapError, SwapErrorType } from 'lib/swapper/api'
+import type { SymbolDenomMapping } from 'lib/swapper/swappers/OsmosisSwapper/utils/helpers'
+import {
+  buildApiTradeTx,
+  buildPerformIbcTransferUnsignedTx,
+  symbolDenomMapping,
+} from 'lib/swapper/swappers/OsmosisSwapper/utils/helpers'
+import { createDefaultStatusResponse } from 'lib/utils/evm'
+
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { atomOnOsmosisAssetId, COSMO_OSMO_CHANNEL } from './utils/constants'
+import type { OsmosisSupportedChainId } from './utils/types'
+
+const tradeQuoteMetadata: Map<string, { chainId: OsmosisSupportedChainId }> = new Map()
+
+export const osmosisApi: Swapper2Api = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    { sellAssetUsdRate }: { sellAssetUsdRate: string },
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    const tradeQuoteResult = await getTradeQuote(input, { sellAssetUsdRate })
+
+    return tradeQuoteResult.map(tradeQuote => {
+      const { receiveAddress, affiliateBps } = input
+      const id = uuid()
+      // TODO(gomes): getTradeQuote() is currently a hack, which represents the swap as a single trade, and needs to be revamped
+      // This effectively means that getUnsignedTx() won't be able to be implemented for all steps, but only the first for now
+      // i.e either the IBC transfer or swap-exact-amount-in
+      tradeQuoteMetadata.set(id, {
+        chainId: tradeQuote.steps[0].sellAsset.chainId as OsmosisSupportedChainId,
+      })
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({
```

Comment:
As mentioned in the PR, this isn't a fully functional implementation but rather an implementation of the first-hop, untested and authored as a base boilerplate which will allow us to iterate on once Osmosis trade quote as a multi-hop is implemented, see the TODO above.